### PR TITLE
Glossary: add workspace and related cli terms

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jbogarthyde** [comment]
Added to CLI description, added terms "workspace", "project", "schematic".

**Comment by jbogarthyde** [comment]
@Brocco Can you check these, and give me a definition of an Angular library?

File: aio/content/guide/glossary.md [hunk 1]
```diff
@@ -677,6 +692,11 @@ The view hierarchy does not imply a component hierarchy. Views that are embedded
 
 See [Custom element](guide/glossary#custom-element)
 
+{@a workspace}
+
+## Workspace
+
+In Angular, a folder that contains a set of related [projects](guide/glossary#project). Angular apps and libraries are always contained in a workspace. The [CLI](guide/glossary#cli) `add` command creates a workspace, and other commands must be executed from within a workspace folder.
```
Comment (Brocco) [inline aio/content/guide/glossary.md]: This is incorrect, the `new` command is what creates a workspace, not the `add` command.

File: aio/content/guide/glossary.md [hunk 2]
```diff
@@ -472,6 +489,11 @@ To learn more, see the [pipes](guide/pipes) page.
 
 An [NPM package](guide/npm-packages) that plugs gaps in a browser's JavaScript implementation. See the [Browser Support](guide/browser-support) guide for polyfills that support particular functionality for particular platforms. 
 
+{@a project}
+
+## Project
+
+In Angular, a folder within a [workspace](guide/glossary#workspace) that contains one or more Angular apps or [libraries](guide/glossary#library).
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: We should get @Brocco to confirm or correct this definition.

**Comment by jbogarthyde** [comment]
@mhevery I have added some CLI terms and updated some DI terms - can you please check them? thanks.

File: aio/content/guide/glossary.md [hunk 3]
```diff
@@ -373,6 +388,17 @@ Lazy loading speeds up application load time by splitting the application into m
 For example, dependencies can be lazy-loaded as needed&emdash;as opposed to "eager-loaded" modules that are required by the root module, and are thus loaded on launch. 
 Similarly, the [router](guide/glossary#router) can load child views only when the parent view is activated, and you can build custom elements that can be loaded into an Angular app when needed.
 
+{@a library}
+
+## Library
+
+In Angular, a [project](guide/glossary#project) that provides functionality that can be included in other Angular apps. A library is not a complete Angular app, and it cannot run independently. 
+
+* Library developers can use the [CLI](guide/glossary#cli) to `generate` a new generic library in an existing [workspace](guide/glossary#workspace), and can publish a library as an `npm` package.
```
Comment (mhevery) [inline aio/content/guide/glossary.md]: i don't think the word `generic` is adding value. Should it be removed?

File: aio/content/guide/glossary.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhevery) [reply aio/content/guide/glossary.md]: +1 for removing `In Angular` as  idon't think it adds value.

File: aio/content/guide/glossary.md [hunk 4]
```diff
@@ -677,6 +723,12 @@ The view hierarchy does not imply a component hierarchy. Views that are embedded
 
 See [Custom element](guide/glossary#custom-element)
 
+{@a workspace}
+
+## Workspace
+
+In Angular, a folder that contains a set of related [projects](guide/glossary#project).
```
Comment (mhevery) [inline aio/content/guide/glossary.md]: `projects`? I think it should be singular. I thought that you can only have one project per workspace.

File: aio/content/guide/glossary.md (hunk 3 above)
Comment (jenniferfell) [reply aio/content/guide/glossary.md]: Maybe we replace "generic" with what we mean by that. empty library? new library folder with default files? new empty library folder? new library with default contents? (playing here)

Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: Using "..to generate scaffolding for a new library"

File: aio/content/guide/glossary.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: I have been using "In Angular" to emphasize that a commonly-used general computing term has a specific meaning in Angular. The term "project" is in very general use for IDEs.

**Comment by jbogarthyde** [comment]
@mhevery @jenniferfell This is all green, I think it is ready to approve.

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**